packages/wonder-blocks-popover/components/close-button.test.js

```diff
@@ -0,0 +1,17 @@
+// @flow
+import * as React from "react";
+
+import expectRenderError from "../../../utils/testing/expect-render-error.js";
+import CloseButton from "./close-button.js";
+import PopoverContext from "./popover-context.js";
+
+describe("CloseButton", () => {
+    test("PopoverContext.Provider and onClose should warn", () => {
+        expectRenderError(
```

Comment:
Hmmm, I don't like test cases that don't follow the AAA layout. It tends to obfuscate what's being tested from what is being asserted.

Could you put the JSX as a variable in the arrange or act, then have the expectation in the assert.

Using shared helper asserts like `expectRenderError` also makes debugging tests hard because the error is thrown in a place unrelated to the test case that fails, which can make the test costly to fix if it starts failing. Although shared code in non-test code is usually a good thing, in tests I believe the opposite is the case as coupling tests together can make it really hard to maintain those tests. Tests should be as isolated as possible.

What do you think about copy pasting the work done by expectRenderError and following AAA layout for consistency and ease of maintenance, rather than sharing code across test cases?

Reply:
Fair enough.

What do you think of this approach?

```
// Arrange
const nodes = (
    <PopoverContext.Provider value={{close: () => {}}}>
        <PopoverContent
            title="title"
            content="content"
            onClose={() => {}}
        />
        ,
    </PopoverContext.Provider>
);

// Act
const underTest = () => mount(nodes);

// Assert
expect(underTest).toThrowError(
    "You've specified 'onClose' on the content when using Popover. Please specify 'onClose' on the Popover instead",
);
```